internal.nix

```diff
@@ -273,7 +273,7 @@ rec {
       getAttr = name: from: lib.optionalAttrs (builtins.hasAttr name from) { "${name}" = from.${name}; };
       getAttrs = names: from: lib.foldl (a: b: a // (getAttr b from)) { } names;
     in
-    (getAttrs [ "src" "nodejs" ] attrs // node_modules_attrs);
+    (getAttrs [ "src" "packageJson" "packageLockJson" "buildInputs" "nativeBuildInputs" "nodejs" "preBuild" "githubSourceHashMap" "passthru" ] attrs // node_modules_attrs);
```

Comment:
I don't think we want to pass through anything that defines a package dependency. I could get on-board with the lockfile location being passed through but that is also already the case for most use cases where you can provide `src`.

The `node_modules` might require different buildInputs and nativeBuildInputs than what you want to have in your regular development shell.

Reply:
thanks, so only `packageLockJson` makes sense